File: backend/common/src/main/java/es/upsa/dasi/tfg/common/domain/model/RolProyecto.java
package es.upsa.dasi.tfg.common.domain.model;

import java.util.Arrays;

public enum RolProyecto
{
    PRODUCT_OWNER("Product Owner"),
    SCRUM_MASTER("Scrum Master"),
    DEVELOPER("Equipo de Desarrollo");

    private final String etiqueta;

    RolProyecto(String etiqueta) {
        this.etiqueta = etiqueta;
    }

    public String getEtiqueta() {
        return etiqueta;
    }

    public static String etiquetaDe(String codigo) {
        if (codigo == null) return "";
        return Arrays.stream(values())
                .filter(r -> r.name().equals(codigo))
                .findFirst()
                .map(RolProyecto::getEtiqueta)
                .orElse(codigo);
    }
}

File: frontend-react/src/utils/rolProyecto.ts
const ETIQUETAS: Record<string, string> = {
  PRODUCT_OWNER: 'Product Owner',
  SCRUM_MASTER: 'Scrum Master',
  DEVELOPER: 'Equipo de Desarrollo',
  TUTOR: 'Tutor',
};

export function etiquetaRol(rol: string, rolEtiqueta?: string): string {
  return rolEtiqueta ?? ETIQUETAS[rol] ?? rol;
}

export const ROLES_INVITACION = ['DEVELOPER', 'SCRUM_MASTER'] as const;

export function miembrosAsignables<T extends { rol: string }>(miembros: T[]): T[] {
  return miembros.filter((m) => m.rol !== 'TUTOR');
}

export function esProductOwner(
  miembros: { usuarioUid: string; rol: string }[],
  usuarioUid?: string | null,
  propietarioUid?: string | null,
): boolean {
  if (!usuarioUid) return false;
  if (propietarioUid && propietarioUid === usuarioUid) return true;
  return miembros.some((m) => m.usuarioUid === usuarioUid && m.rol === 'PRODUCT_OWNER');
}
